_posts/2021-11-02-choosing-dask-chunk-sizes.md

```diff
@@ -0,0 +1,157 @@
+- Lower bound: To get the advantage of parallization, you need the number of chunks to at least equal the number of workers available (or better, the number of workers times 2). Otherwise, if the number of chunks is less than the number of workers, some workers will stay idle.
+- The time taken to compute each task should be much larger than the time needed to schedule the task. The Dask scheduler takes roughly 1 millisecond to coordinate a single task, so a good task computation time would be measured in seconds (not milliseconds).
+
+### Chunks should be aligned with array storage on disk
+
+If you are reading data from disk, the storage structure will inform what shape your Dask array chunks should be. For best performance, choose chunks that are well aligned with the way data is stored.
+
+From the Dask best practices on how to [orient your chunks](https://docs.dask.org/en/stable/array-best-practices.html#orient-your-chunks):
+
+> When reading data you should align your chunks with your storage format. Most array storage formats store data in chunks themselves. If your Dask array chunks aren’t multiples of these chunk shapes then you will have to read the same data repeatedly, which can be expensive. Note though that often storage formats choose chunk sizes that are much smaller than is ideal for Dask, closer to 1MB than 100MB. In these cases you should choose a Dask chunk size that aligns with the storage chunk size and that every Dask chunk dimension is a multiple of the storage chunk dimension.
+
+Some examples of data storage structures on disk include:
+- A HDF5 or [Zarr array](https://zarr.readthedocs.io/en/stable/api/core.html). The size and shape of chunks/blocks stored on disk should align well with the Dask array chunks you select.
+- A folder full of tiff files. You might decide that each tiff file should become a single chunk in the Dask array (or that multiple tiff files should be grouped into a single chunk).
+
+### Rechunking arrays
+
+If you want to change the chunking of a Dask array, you can do that with the [rechunk](https://docs.dask.org/en/latest/generated/dask.array.rechunk.html) method.
+
+```python
+rechunked_array = original_array.rechunk(new_shape)
+```
+
+**Warning:** Rechunking Dask arrays comes at a cost.
+- The Dask graph must be rearranged to accomodate the new chunk structure. This happens immediately, and will block any other interaction with python until Dask has rearranged the task graph.
+- This also inserts new tasks into the Dask graph. At compute time, there are now more tasks to execute.
+
+For these reasons, it is best to avoid rechunking. However, sometimes the data is stored on disk is not well aligned and rechunking may be necessary.
+For an example of this, here is Draga Doncila Pop [talking about chunk alignment](https://youtu.be/10Ws59NGDaE?t=833) with satellite image data.
+
+## Too small is a problem
```

Comment:
I'll would put this part and the one after just below the "How do I nkow what chunks my array has?" part, or even just before.

Reply:
Section order rearranged. I think that is better, thank you